### PR TITLE
chore(otelcol): upgrade to 0.0.58-beta.0

### PR DESCRIPTION
**Pull request by sumo-drosiek**
##### Description

upgrade OTC to 0.0.58-beta.0
change memory limit for otc to be 75% of available memory with 20% spike
 
---

##### Checklist

- [ ] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - chore(deps): bump prometheus node exporter tag to 2.3.1 [#2177][#2177]
 - chore: upgrade Fluentd to 1.14.5-sumo-1 [#2196][#2196]
 - chore: upgrade Falco Helm Chart to 1.17.4 [#2197][#2197]
+- chore: bump sumo ot distro to 0.0.50-beta.0 [#2213][#2213]
```
Comment (pmalek-sumo) [inline CHANGELOG.md:25]:
```suggestion
- chore: bump sumo ot distro to 0.0.58-beta.0 [#2213][#2213]
```

?

**Comment by pmalek-sumo** [comment]
> 🤔
> 
> ```
> Test_Helm_Traces_Load_Balancing 2022-03-31T11:49:55Z logger.go:66: pod/rel-2845692050-sumologic-otelgateway-784499b9dd-h9q76   0/1     CrashLoopBackOff   4 (44s ago)   2m15s
> ```

```
2022-04-01T09:49:26.208Z    info    memorylimiterprocessor/memorylimiter.go:140    Using percentage memory limiter    {"kind": "processor", "name": "memory_limiter", "pipeline": "metrics", "total_memory_mib": 2048, "limit_percentage": 75, "spike_limit_percentage": 0}
Error: cannot build pipelines: error creating processor "memory_limiter" in pipeline "metrics": memoryLimitPercentage and memorySpikePercentage must be greater than zero and less than or equal to hundred
2022/04/01 09:49:26 collector server run finished with error: cannot build pipelines: error creating processor "memory_limiter" in pipeline "metrics": memoryLimitPercentage and memorySpikePercentage must be greater than zero and less than or equal to hundred
```

**Comment by sumo-drosiek** [comment]
`memorySpikePercentage` - what do we want to set for it? `95%` ?

**Comment by pmalek-sumo** [comment]
> `spike_limit_percentage` (default = 0): Maximum spike expected between the measurements of memory usage. The value must be less than limit_percentage. This option is used to calculate spike_limit_mib from the total available memory. For instance setting of 25% with the total memory of 1GiB will result in the spike limit of 250MiB. This option is intended to be used only with limit_percentage.

I believe rather something like 20%.

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**